Makefile

```diff
@@ -8,6 +8,21 @@ PRE_RELEASE := "1"
 RELEASE_TYPE := $(shell if [ $(PRE_RELEASE) = "0" ] ; then echo release; else echo pre-release ; fi)
 GOLANGCI_BINARY=golangci-lint
 
+ifeq ($(OS),Windows_NT)     # is Windows_NT on XP, 2000, 7, Vista, 10...
+ detected_OS := Windows
+else
+ detected_OS := $(strip $(shell uname))
+endif
+
+ifeq ($(detected_OS),Darwin)
+ GOBIN_SHARED_LIB_EXT := dylib
+else ifeq ($(detected_OS),Windows)
+ GOBIN_SHARED_LIB_CGO_LDFLAGS := CGO_LDFLAGS="-Wl,--export-all-symbols"
```

Comment:
Why is this necessary only for windows?

Reply:
I discovered the need for that linker flag on Windows a few months ago when attempting to hook status-go (built as a shared library) to Clojure via JNA.  Without it, *none* of the expected symbols could be found in `libstatus.dll`. When creating this target for the sake of nim-status/-client, I tried compiling for Windows without the flag and had the same problem.

I searched and experimented for hours back in May before stumbling on this solution in some Stack Overflow comment (didn't save the link unfortunately). There may be a better way, but it works and I'm not aware of any downsides.

Reply:
Can we add a comment explaining why that's there then? Or explain it in the commit message?

Reply:
Sure, I'll reword the commit message to include an explanation.

Reply:
Done.

Reply:
Nice.